test/unit/transactionBodyTest.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-env mocha */
+
+import { retrieveTransactionBody } from '../../src/middleware/transactionBody'
+import uuid from 'uuid'
+import mongodb from 'mongodb'
+import { config } from '../../src/config'
+import fs from 'fs'
+
+describe('retrieveTransactionBody()', async () => {
+  let db
+  let client
+  const mongoClient = mongodb.MongoClient
+
+  before(async () => {
+    client = await mongoClient.connect(config.mongo.url)
```

Comment:
We dont have to create a new mongoose connection, but can instead use an existing mongoose connection to connect to.
https://github.com/jembi/openhim-core-js/pull/1019/files#diff-f8e8ecebde8954f0b9cebb0a9c5c9202R7

We can the define our before/after hooks for setting up/cleaning up the database connection
https://github.com/jembi/openhim-core-js/pull/1019/files#diff-f8e8ecebde8954f0b9cebb0a9c5c9202R11
https://github.com/jembi/openhim-core-js/pull/1019/files#diff-f8e8ecebde8954f0b9cebb0a9c5c9202R16

The for each test, we can clean out the collections to ensure a clean slate for the next text
https://github.com/jembi/openhim-core-js/pull/1019/files#diff-f8e8ecebde8954f0b9cebb0a9c5c9202R21